crypto/key_manager.go

```diff
@@ -24,50 +24,27 @@ import (
 	"fmt"
 	"io/ioutil"
 
-	"github.com/golang/glog"
 	"github.com/google/trillian/crypto/sigpb"
 )
 
-// KeyManager loads and holds our private and public keys. Should support ECDSA and RSA keys.
-// The crypto.Signer API allows for obtaining a public key from a private key but there are
-// cases where we have the public key only, such as mirroring another log, so we treat them
-// separately. KeyManager is an interface as we expect multiple implementations supporting
-// different ways of accessing keys.
-type KeyManager interface {
-	// Signer returns a crypto.Signer that can sign data using the private key held by the
-	// manager.
-	Signer() (crypto.Signer, error)
+// PrivateKeyManager supports signing data with a private key that may be stored
+// in a secure location, which is not immediately available to this client.
+type PrivateKeyManager interface {
+	// Signer returns a crypto.Signer that can sign data using the private key.
+	Signer() crypto.Signer
 	// SignatureAlgorithm returns the value that identifies the signature algorithm.
 	SignatureAlgorithm() sigpb.DigitallySigned_SignatureAlgorithm
 	// HashAlgorithm returns the type of hash that will be used for signing with this key.
 	HashAlgorithm() crypto.Hash
-	// GetPublicKey returns the public key previously loaded. It is an error to call this
-	// before a public key has been loaded
-	GetPublicKey() (crypto.PublicKey, error)
-	// GetRawPublicKey returns the DER encoded public key bytes.
-	// This is needed for some applications that exchange or embed key hashes in structures.
-	// It is an error to call this before a public key has been loaded
-	GetRawPublicKey() ([]byte, error)
+	// PublicKey returns the public key corresponding to the private key.
+	PublicKey() crypto.PublicKey
 }
 
 // PEMKeyManager is an instance of KeyManager that loads its key data from an encrypted
 // PEM file.
 type PEMKeyManager struct {
```

Comment:
I don't think PEMKeyManager actually has anything to do with PEM any more. All of the PEM processing has been moved to free functions. Consider renaming it.